tensorflow/core/grappler/optimizers/auto_mixed_precision_test.cc

```diff
@@ -1139,110 +1203,119 @@ TEST_F(AutoMixedPrecisionTest, BatchMatMul) {
+TEST_P(AutoMixedPrecisionParamTest, LogOp) {
   TestSimpleUnaryInferOp(
       0.01, 10, 1.0e-3, 2.0e-3,
       [](const tensorflow::Scope& scope, Output input) -> Output {
         return ops::Log(scope, input);
-      });
+      }, GetParam());
 }
 
-TEST_F(AutoMixedPrecisionTest, Log1pOp) {
+TEST_P(AutoMixedPrecisionParamTest, Log1pOp) {
   TestSimpleUnaryInferOp(
       -0.99, 9, 1.0e-3, 5.0e-3,
       [](const tensorflow::Scope& scope, Output input) -> Output {
         return ops::Log1p(scope, input);
-      });
+      }, GetParam());
 }
 
-TEST_F(AutoMixedPrecisionTest, LogSoftmaxOp) {
+TEST_P(AutoMixedPrecisionParamTest, LogSoftmaxOp) {
   TestSimpleUnaryInferOp(
       -8, 8, -1, 1.0e-2,
       [](const tensorflow::Scope& scope, Output input) -> Output {
         return ops::LogSoftmax(scope, input);
-      });
+      }, GetParam());
 }
 
-TEST_F(AutoMixedPrecisionTest, ReciprocalOp) {
+TEST_P(AutoMixedPrecisionParamTest, ReciprocalOp) {
   TestSimpleUnaryInferOp(
       0.01, 10, -1, 1.0e-3,
       [](const tensorflow::Scope& scope, Output input) -> Output {
         return ops::Reciprocal(scope, input);
-      });
+      }, GetParam());
 }
 
-TEST_F(AutoMixedPrecisionTest, SigmoidOp) {
+TEST_P(AutoMixedPrecisionParamTest, SigmoidOp) {
   TestSimpleUnaryInferOp(
       -5, 5, 1.0e-3, -1,
       [](const tensorflow::Scope& scope, Output input) -> Output {
         return ops::Sigmoid(scope, input);
-      });
+      }, GetParam());
 }
 
-TEST_F(AutoMixedPrecisionTest, SoftmaxOp) {
+TEST_P(AutoMixedPrecisionParamTest, SoftmaxOp) {
   TestSimpleUnaryInferOp(
       -8, 8, 2.0e-3, -1,
       [](const tensorflow::Scope& scope, Output input) -> Output {
         return ops::Softmax(scope, input);
-      });
+      }, GetParam());
 }
 
-TEST_F(AutoMixedPrecisionTest, SoftplusOp) {
+TEST_P(AutoMixedPrecisionParamTest, SoftplusOp) {
   TestSimpleUnaryInferOp(
       -5, 5, 2.0e-3, 2.0e-3,
       [](const tensorflow::Scope& scope, Output input) -> Output {
         return ops::Softplus(scope, input);
-      });
+      }, GetParam());
 }
 
-TEST_F(AutoMixedPrecisionTest, SqrtOp) {
+TEST_P(AutoMixedPrecisionParamTest, SqrtOp) {
   TestSimpleUnaryInferOp(
       0, 10, 1.0e-3, 1.0e-3,
       [](const tensorflow::Scope& scope, Output input) -> Output {
         return ops::Sqrt(scope, input);
-      });
+      }, GetParam());
 }
 
-TEST_F(AutoMixedPrecisionTest, TanhOp) {
+TEST_P(AutoMixedPrecisionParamTest, TanhOp) {
   TestSimpleUnaryInferOp(
       -5, 5, 1.0e-3, -1,
       [](const tensorflow::Scope& scope, Output input) -> Output {
         return ops::Tanh(scope, input);
-      });
+      }, GetParam());
 }
 
+INSTANTIATE_TEST_SUITE_P(AutoMixedPrecisionTest, AutoMixedPrecisionParamTest,
+                         ::testing::ValuesIn({
+#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
+                           AutoMixedPrecisionMode::CUDA,
+#elif INTEL_MKL
+                           AutoMixedPrecisionMode::FP16_CPU
+#endif
```

Comment:
If compiled for both CUDA and MKL, I think we can run both sets of tests. So perhaps this should be:

```
#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
                           AutoMixedPrecisionMode::CUDA,
#endif
#if INTEL_MKL
                           AutoMixedPrecisionMode::FP16_CPU,
#endif
```

Reply:
done.